### ISSUE TITLE
Lua tp() rollback example error

### ISSUE THREAD

**Issue by jsikri94** [orbitalquark/lua-yottadb#6]
Test program for tp():
```
local ydb = require('yottadb')

function f()
   return ydb.YDB_TP_ROLLBACK
end

ydb.tp('', {'*'}, f)
```

Output:
```
lua: YDB Error: 2147483645: 
stack traceback:
        [C]: in function '_yottadb.tp'
        ./yottadb.lua:390: in function 'yottadb.tp'
        tests/test_tp.lua:25: in main chunk
        [C]: in ?
```

Is this the expected output?

**Comment by orbitalquark** [comment]
Yes, you are expected to use Lua's `pcall()` with `yottadb.tp()` if there is a possibility of rollback, restart, or some other error.

https://github.com/orbitalquark/lua-yottadb/blob/28073b87ce2a707d7e722e11bbbc608100875a69/tests/test.lua#L661

**Comment by jsikri94** [comment]
Ah, okay.

Would something like this be correct?

```
function increment_global()
   print("Transaction starting...")
   local ok, e = pcall(ydb.incr, '^num', 5) -- increment by 1
   if (not ok) then
      if (ydb.get_error_code(e) == ydb.YDB_TP_RESTART) then
      	 return ydb.YDB_TP_RESTART
      end
      if (ydb.get_error_code(e) == ydb.YDB_TP_ROLLBACK) then
      	 return ydb.YDB_TP_ROLLBACK
      end
   end
end

ydb.tp('', {'*'}, increment_global)
```

**Comment by orbitalquark** [comment]
I don't think so. `ydb.incr()` would not return a RESTART or ROLLBACK error as far as I can see from your code, so the error code tests are useless. The `ydb.tp()` call would probably not error in this case, but I don't think it would be doing what you expect.

**Comment by ksbhaskar** [comment]
For what it's worth `ydb.incr()` can return a restart, because of the optimistic concurrency control implementation in YottaDB, but not a rollback.

**Comment by orbitalquark** [comment]
Okay, thanks for the correction. Then you'd probably just wrap `ydb.tp()` in a `pcall` to gracefully handle a rollback and have Lua continue instead of error out.

**Comment by jsikri94** [comment]
Done. Thank you both.

**Comment by orbitalquark** [comment]
Yes, you are expected to use Lua's `pcall()` with `yottadb.tp()` if there is a possibility of rollback, restart, or some other error.

https://github.com/orbitalquark/lua-yottadb/blob/28073b87ce2a707d7e722e11bbbc608100875a69/tests/test.lua#L661\
